### PR TITLE
NULL dereference fixes

### PR DESCRIPTION
**Pull request by akallabeth**
Fixed a few missing checks and code cleanup.

### CODE REVIEW

**Comment by mfleisz** [comment]
+1

**Comment by bmiklautz** [comment]
+1 for the reallocs
~1 for the invalid handles - a lot handle specific functions return NULL and don't really use INVALID_HANDLE_VALUE. Using INVALID_HANDLE_VALUE in those cases might cause problems since it's theoratically valid memory that can be returned. See also http://blogs.msdn.com/b/oldnewthing/archive/2004/03/02/82639.aspx
+1 for the formats check. might not be necessary but also doesn't hurt ;)

**Comment by mfleisz** [comment]
Yes we should only check INVALDI_HANDLE_VALUE for the APIs that use that return value (i.e. the File API) and skip the check on the others (i.e. Event/Mutex API)

**Comment by akallabeth** [comment]
@bmiklautz @mfleisz Removed the ```INVALID_HANDLE_VALUE``` check from handle types not using it.

**Comment by mfleisz** [comment]
@akallabeth Event/Mutex/Semaphore functions don't use INVALID_HANDLE_VALUE either.

**Comment by akallabeth** [comment]
@mfleisz ah, correct. Mixed ```INVALID_HANDLE_VALUE``` with ```ERROR_INVALID_HANDLE```

File: winpr/libwinpr/synch/timer.c [hunk 1]
```diff
@@ -49,7 +49,7 @@ static BOOL TimerIsHandled(HANDLE handle)
 {
 	WINPR_TIMER* pTimer = (WINPR_TIMER*) handle;
 
-	if (!pTimer || pTimer->Type != HANDLE_TYPE_TIMER)
+	if (!pTimer || (pTimer->Type != HANDLE_TYPE_TIMER) || (pTimer == INVALID_HANDLE_VALUE))
 	{
```
Comment (nfedera) [inline winpr/libwinpr/synch/timer.c]: Don't think INVALID_HANDLE_VALUE is possible here, please check again

File: winpr/libwinpr/utils/wlog/wlog.c [hunk 2]
```diff
@@ -353,7 +353,8 @@ int WLog_ParseFilter(wLogFilter* filter, LPCSTR name)
 
 	while ((p = strchr(p, '.')) != NULL)
 	{
-		filter->Names[count++] = p + 1;
+		if ((p + 1) && (count < filter->NameCount))
+			filter->Names[count++] = p + 1;
```
Comment (nfedera) [inline winpr/libwinpr/utils/wlog/wlog.c]: I guess (p+1) should be *(p+1) or p[1]

File: winpr/libwinpr/utils/wlog/wlog.c [hunk 3]
```diff
@@ -380,34 +381,39 @@ int WLog_ParseFilters()
...
-	while ((p = strchr(p, ',')) != NULL)
-	{
-		strs[count++] = p + 1;
-		*p = '\0';
-		p++;
+		strs[count++] = p;
+
+		while ((p = strchr(p, ',')) != NULL)
+		{
+			if ((p + 1) && (count < g_FilterCount))
```
Comment (nfedera) [inline winpr/libwinpr/utils/wlog/wlog.c]: again: p + 1 is always true, probably that should be if (p[1] ...)  or  if (*(p+1) ...)

File: winpr/libwinpr/utils/wlog/wlog.c [hunk 4]
```diff
@@ -502,7 +508,8 @@ int WLog_ParseName(wLog* log, LPCSTR name)
 
 	while ((p = strchr(p, '.')) != NULL)
 	{
-		log->Names[count++] = p + 1;
+		if ((p + 1) && (count < log->NameCount))
```
Comment (nfedera) [inline winpr/libwinpr/utils/wlog/wlog.c]: again: p + 1 is always true, see above

**Comment by nfedera** [comment]
+1 after fixing my remarks. thanks !

**Comment by akallabeth** [comment]
@nfedera Fixed your remarks, the ```p+1``` was actually ```p[0]``` as in check for end of string to abort.

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by bmiklautz** [comment]
@freerdp-bot test

File: winpr/libwinpr/utils/wlog/wlog.c [hunk 5]
```diff
@@ -380,34 +381,39 @@ int WLog_ParseFilters()
...
 
-	if (!strs)
-	{
-		free(env);
-		return -1;
+		g_FilterCount = count;
+		p = env;
 	}
+	{
+		count = 0;
```
Comment (bmiklautz) [inline winpr/libwinpr/utils/wlog/wlog.c]: Why are there to extra scopes ({}) used there?

**Comment by bmiklautz** [comment]
If the unnecessary scopes are removed +1.

**Comment by bmiklautz** [comment]
@freerdp-bot test.

**Comment by bmiklautz** [comment]
+1

**Comment by bmiklautz** [comment]
I've created a re-based version as #2509

**Pull request closed without merging**